x-pack/plugins/security_solution/server/endpoint/routes/actions/response_actions.ts

```diff
@@ -0,0 +1,312 @@
+      try {
+        logsEndpointActionsResult = await esClient.index<LogsEndpointAction>(
+          {
+            index: `${ENDPOINT_ACTIONS_DS}-default`,
+            body: {
+              ...doc,
+            },
+            refresh: 'wait_for',
+          },
+          { meta: true }
+        );
+        if (logsEndpointActionsResult.statusCode !== 201) {
+          return res.customError({
+            statusCode: 500,
+            body: {
+              message: logsEndpointActionsResult.body.result,
+            },
+          });
+        }
+      } catch (e) {
+        return res.customError({
+          statusCode: 500,
+          body: { message: e },
+        });
+      }
+    }
+
+    // write actions to .fleet-actions index
+    try {
+      fleetActionIndexResult = await esClient.index<EndpointAction>(
+        {
+          index: AGENT_ACTIONS_INDEX,
+          body: {
+            ...doc.EndpointActions,
+            '@timestamp': doc['@timestamp'],
+            agents,
+            timeout: 300, // 5 minutes
+            user_id: doc.user.id,
+          },
+          refresh: 'wait_for',
+        },
+        { meta: true }
+      );
+
+      if (fleetActionIndexResult.statusCode !== 201) {
+        return res.customError({
+          statusCode: 500,
+          body: {
+            message: fleetActionIndexResult.body.result,
+          },
+        });
+      }
+    } catch (e) {
+      // create entry in .logs-endpoint.action.responses-default data stream
+      // when writing to .fleet-actions fails
+      if (doesLogsEndpointActionsDsExist) {
+        await createFailedActionResponseEntry({
+          context,
+          doc: {
+            '@timestamp': moment().toISOString(),
+            agent: doc.agent,
+            EndpointActions: {
+              action_id: doc.EndpointActions.action_id,
+              completed_at: moment().toISOString(),
+              started_at: moment().toISOString(),
+              data: doc.EndpointActions.data,
+            },
+          },
+          logger,
+        });
+      }
+      return res.customError({
+        statusCode: 500,
+        body: { message: e },
+      });
+    }
+
+    // Update all cases with a comment
+    if (caseIDs.length > 0) {
+      const targets = endpointData.map((endpt: HostMetadata) => ({
+        hostname: endpt.host.hostname,
+        endpointId: endpt.agent.id,
+      }));
+
+      await Promise.all(
+        caseIDs.map((caseId) =>
+          casesClient.attachments.add({
+            caseId,
+            comment: {
+              type: CommentType.actions,
+              comment: req.body.comment || '',
+              actions: {
+                targets,
+                type: command,
+              },
+              owner: APP_ID,
+            },
+          })
+        )
+      );
+    }
+
+    return res.ok({
+      body: {
+        action: actionID,
+        actionDetails: await getActionDetailsById(esClient, actionID),
```

Comment:
now additionally returning `ActionDetails`

Reply:
@paul-tavares is this right?

Reply:
I have not looked at this PR, but I think we agreed at the Eng. office hour that we would spread the `actionDetails` into the response so that we could maintain backward compatibility with Isolate/Release.

Also,
since this is a generic handler for all actions, I would not return `action` on every action - only for `release` and `isolate`. no point in making all apis for action have this extra property in it

Reply:
👌  on the `action` only on isolate/release. as for spreading the `actionDetails`, wouldn't keeping it nested be a little more future proof should we need to add another property that's not a part of `actionDetails` in the future?

Reply:
**TL;DR;** -- I'm ok if you all agree on returning `{ action: string; actionDetails: ActionDetails }`. (but maybe just name the second property `details`?

_____
Our normal (or agreed upon shape) API return response should have a property named `data` that includes the response's payload and not the actual record (in the case `ActionDetails`), so it would look like this:

```Typescript
{
    data: ActionDetails
}
```

this response shape lends itself to having additional data be returned in the future, but doing this now would impact `isolate` and `release`, so we said that were just going to return the `ActionDetails` and since there was no overlap with `release`/`isolate`, you could just spread it.

Reply:
🤔 actually... naming the new property `data` and having it value be `ActionDetails` actually moves us closer to the standard response (and would match the action details api)

Reply:
so like this?
```
{
	action: string // if isolate or release
	data: ActionDetails
}
```

Reply:
yeah - that 👍

Reply:
updated to above